### PR TITLE
Feature/2020 11 12 typescript in a serverless backend with NestJS and Azure Functions

### PR DESCRIPTION
**Pull request by jasperrosiers**
### Thank you for your contribution! Please go through this small checklist to prepare your pull request to be reviewed as soon as possible :)
* [x] All authors were added to `_data/authors.yml` with a picture and an up-to-date bio
* [x] The blog post image is of ratio 500x293
  * Feel free to leave a comment if you don't know how to resize your image and someone will gladly help you out
* [x] The [**one sentence per line** method](https://raw.githubusercontent.com/brandon-rhodes/blog/master/texts/brandon/2012/one-sentence-per-line.rst) has been adopted
* [x] A spell check has been done
* [x] All links open in a new tab
  * Example: `[Amazing tech blog](https://ordina-jworks.github.io){:target="_blank" rel="noopener noreferrer"}`
* [x] The post's content properly resizes according to the window's size
  * Tip: Just shorten your window's horizontal length to easily test this
  * To have images properly resize use `class="image fit"`
  * To have YouTube videos properly resize, wrap the iframe into a `<div class="responsive-video">`
* [x] If a table of content is: check whether all links are working correctly

### Any specific remarks for the reviewers or anything you need help with?
\<enter text here>

### Checklist for the reviewers before merging
* [x] Publishing date of the post has been correctly set
  * This can be done by renaming the filename or by setting the date in the header section of the post

### CODE REVIEW

**Review by DuncanCasteleyn — APPROVED** [review]
Looking good 🎉

File: _posts/2020-11-12-Typescript-in-a-serverless-backend-with-NestJS-and-Azure-Functions.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+  ]
+})
+export class ParametersModule {}
+```
+
+The first step when receiving an HTTP request to the application is the `ParametersController` as shown below.
+This controller will catch all requests on the 'parameters' endpoint.
+Using annotations, you can:
+- make a check for the type of incoming request and divide the traffic accordingly. This is similar to the way annotations work in the spring framework (eg. `@PostMapping()`).
+- customise which Http code you want to return on successful calls, as I did with the `createParameters` method.
```
Comment (yannickdeturck) [inline _posts/2020-11-12-Typescript-in-a-serverless-backend-with-NestJS-and-Azure-Functions.md:96]: I'd write it as HTTP to be consistent with the other usages throughout the blogpost.
```suggestion
- Customise which HTTP code you want to return on successful calls, as I did with the `createParameters` method.
```

File: _posts/2020-11-12-Typescript-in-a-serverless-backend-with-NestJS-and-Azure-Functions.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+## Authorization Guards
+
+The project uses two guards, an `AuthGuard` for the authorization, and a `RolesGuard` to check which roles can access certain resources.
+A good explanation of both can be found in the [NestJS documentation](https://docs.nestjs.com/guards){:target="_blank" rel="noopener noreferrer"}.
+The `RolesGuard` is almost an exact copy from the documentation, so let's take a look at the `AuthGuard` which doesn't need to be provided from a module.
+
+The `canActivate()` method is called before executing the method in the controller.
+It needs to return `true`, or the method won't execute and the application will return a 401 Unauthorized code.
+In this case, we check if the authorization header has the correct value as configured in the environment variables.
+Other setups, like OAuth, Cognito or jwt tokens are also possible.
```
Comment (yannickdeturck) [inline _posts/2020-11-12-Typescript-in-a-serverless-backend-with-NestJS-and-Azure-Functions.md:222]:
```suggestion
Other setups, like OAuth, Cognito or JWT tokens are also possible.
```

File: _posts/2020-11-12-Typescript-in-a-serverless-backend-with-NestJS-and-Azure-Functions.md [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+```
+
+That's it!
+Your app is now fully functional!
+
+## Conclusion
+
+In this blog post, we made a small application to discover how NestJS can be used in the backend with some of its neat features.
+Of course, this was a very basic program to show some of the possibilities.
+For more information on NestJS and its features, check out the very thorough [documentation](https://docs.nestjs.com/){:target="_blank" rel="noopener noreferrer"}
```
Comment (yannickdeturck) [inline _posts/2020-11-12-Typescript-in-a-serverless-backend-with-NestJS-and-Azure-Functions.md:310]: Missing a period at the end. :-)
```suggestion
For more information on NestJS and its features, check out the very thorough [documentation](https://docs.nestjs.com/){:target="_blank" rel="noopener noreferrer"}.
```

File: _posts/2020-11-12-Typescript-in-a-serverless-backend-with-NestJS-and-Azure-Functions.md [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+
+## Resources
+
+- [The body parameters project](https://github.com/jasperrosiers/body-parameter-tracker){:target="_blank" rel="noopener noreferrer"}
```
Comment (yannickdeturck) [inline _posts/2020-11-12-Typescript-in-a-serverless-backend-with-NestJS-and-Azure-Functions.md:314]: Feel free to include your GitHub project again here. 👍🏼

**Review by yannickdeturck — CHANGES_REQUESTED** [review]
Looks good, @jasperrosiers ! Reads pretty fluently. 👌🏼
I've added a couple of suggestions to polish the post a bit further.

**Review by yannickdeturck — APPROVED** [review]
Solid work, @jasperrosiers. 👌🏼
Off it goes! 🚀

**Pull request merged**